corehq/apps/api/resources/messaging_event/utils.py

```diff
@@ -74,10 +74,11 @@ def get_request_params(request):
     """Get the query params from the request. If the request is using a cursor
     decode the cursor and return those values as the query params.
     :returns: CursorParams tuple"""
+    has_cursor = False
     request_params = request.GET.dict()
     if 'cursor' in request_params:
-        params_string = b64decode(request_params['cursor']).decode('utf-8')
-        return CursorParams(
-            QueryDict(params_string).dict(), request.domain, True
-        )
-    return CursorParams(request_params, request.domain, False)
+        has_cursor = True
+        cursor_params_dict = QueryDict(b64decode(request_params['cursor']).decode('utf-8')).dict()
+        request_params.pop('cursor')
+        request_params = {**request_params, **cursor_params_dict}
```

Comment:
It would just make it easier to work with when reading the given parameters. Looking at this again, I think I should maybe revert this, since 1. the `messaging_events` also uses this and 2. the previous behaviour was to either return only the cursor, or the "other" params. I haven't tested the messaging AP, but don't want to mess with that behaviour. Let me change it back..

Reply:
https://github.com/dimagi/commcare-hq/pull/33380/commits/eafd81288e469724f347f317391e40284ccec6f3